tty/pc-tty/ttypc_vtf.c

```diff
@@ -1,617 +1,639 @@
-	(BG_LIGHTGREY | FG_BLACK),                /* invers               */
-	(BG_LIGHTGREY | FG_BLACK),                /* bold+invers          */
-	(BG_LIGHTGREY | FG_BLACK),                /* underline+invers     */
-	(BG_LIGHTGREY | FG_BLACK),                /* bold+underline+invers*/
-	(BG_LIGHTGREY | FG_BLACK | FG_BLINK),     /* blink+invers         */
-	(BG_LIGHTGREY | FG_BLACK | FG_BLINK),     /* bold+blink+invers    */
-	(BG_LIGHTGREY | FG_BLACK | FG_BLINK),     /* underline+blink+invers */
-	(BG_LIGHTGREY | FG_BLACK | FG_BLINK)      /* bold+underl+blink+invers */
+
+/* Monochrome display SGR attributes */
+static uint8_t msgr[] = {
+	(BG_BLACK     | FG_LIGHTGREY),            /* Normal */
+	(BG_BLACK     | FG_UNDERLINE),            /* Bold */
+	(BG_BLACK     | FG_UNDERLINE),            /* Underline */
+	(BG_BLACK     | FG_UNDERLINE),            /* Bold + underline */
+	(BG_BLACK     | FG_LIGHTGREY | FG_BLINK), /* Blink */
+	(BG_BLACK     | FG_UNDERLINE | FG_BLINK), /* Bold + blink */
+	(BG_BLACK     | FG_UNDERLINE | FG_BLINK), /* Underline + blink */
+	(BG_BLACK     | FG_UNDERLINE | FG_BLINK), /* Bold + underline + blink */
+	(BG_LIGHTGREY | FG_BLACK),                /* Inversed */
+	(BG_LIGHTGREY | FG_BLACK),                /* Bold + inversed */
+	(BG_LIGHTGREY | FG_BLACK),                /* Underline + inversed */
+	(BG_LIGHTGREY | FG_BLACK),                /* Bold + underline + inversed */
+	(BG_LIGHTGREY | FG_BLACK     | FG_BLINK), /* Blink + inversed */
+	(BG_LIGHTGREY | FG_BLACK     | FG_BLINK), /* Bold + blink + inversed */
+	(BG_LIGHTGREY | FG_BLACK     | FG_BLINK), /* Underline + blink + inversed */
+	(BG_LIGHTGREY | FG_BLACK     | FG_BLINK)  /* Bold + underline + blink + inversed */
 };
 
 
-/* foreground ANSI color -> pc */
-uint8_t fgansitopc[] = {
-	FG_BLACK, FG_RED, FG_GREEN, FG_BROWN, FG_BLUE,
-	FG_MAGENTA, FG_CYAN, FG_LIGHTGREY
+/* Internal attributes indexes */
+enum {
+	VT_NORMAL   = 0,
+	VT_BOLD     = 1,
+	VT_UNDER    = 2,
+	VT_BLINK    = 4,
+	VT_INVERSED = 8
 };
 
 
-/* background ANSI color -> pc */
-uint8_t bgansitopc[] = {
-	BG_BLACK, BG_RED, BG_GREEN, BG_BROWN, BG_BLUE,
-	BG_MAGENTA, BG_CYAN, BG_LIGHTGREY
+/* Foreground ANSI color code to PC conversion table */
+static uint8_t fgansitopc[] = {
+	FG_BLACK,    /* 0 */
+	FG_RED,      /* 1 */
+	FG_GREEN,    /* 2 */
+	FG_BROWN,    /* 3 */
+	FG_BLUE,     /* 4 */
+	FG_MAGENTA,  /* 5 */
+	FG_CYAN,     /* 6 */
+	FG_LIGHTGREY /* 7 */
 };
 
 
-/* initialize ANSI escape sequence parameter buffers */
-void _ttypc_vtf_clrparms(ttypc_virt_t *virt)
-{
-	register int i;
+/* Background ANSI color code to PC conversion table */
+static uint8_t bgansitopc[] = {
+	BG_BLACK,    /* 0 */
+	BG_RED,      /* 1 */
+	BG_GREEN,    /* 2 */
+	BG_BROWN,    /* 3 */
+	BG_BLUE,     /* 4 */
+	BG_MAGENTA,  /* 5 */
+	BG_CYAN,     /* 6 */
+	BG_LIGHTGREY /* 7 */
+};
 
-	for (i = 0; i < MAXPARMS; i++)
-		virt->parms[i] = 0;
-	virt->parmi = 0;
-}
 
+/* ASCII character set */
+static uint16_t ascii[] = {
```

Comment:
Why do we need table to perform y = x + 0x20 function?

Reply:
The function below is being used for rendering characters onto the VGA screen using appropriate character set (GL ang GR may point to ascii table or any other character set table). As you can see the general approach is to use a table to resolve characters.
I guess we could use functions (Gs, GL and GR would become function pointers) for resolving characters instead. This way we would get rid of unnecessary tables.
```
/* Writes character to screen buffer */
static void _ttypc_vt_sdraw(ttypc_vt_t *vt, char c)
{
	if ((c >= 0x20) && (c <= 0x7f))
		c = (vt->ss) ? (*vt->Gs)[c - 0x20] : (*vt->GL)[c - 0x20];
	else if (c >= 0xa0)
		c = (*vt->GR)[c - 0xa0];

	*(vt->vram + vt->cpos) = vt->attr | c;
	vt->ss = 0;
}
```

Reply:
Ok, this is a valid point imho. No need for change